server/src/main/java/org/opensearch/gateway/BaseGatewayShardAllocator.java

```diff
@@ -132,4 +138,70 @@ protected static List<NodeAllocationResult> buildDecisionsForAllNodes(ShardRouti
         }
         return results;
     }
+
+    protected static class NodeShardState {
```

Comment:
Can you write some comment which explains both 

**NodeShardState** and NodeShardState**s**

Reply:
Added explanation for both